### ISSUE TITLE
Generate Release Notes (Crossplatform) - Allow scripts to access the OAuth token

### ISSUE THREAD

**Issue by ChristoBothaCBS** [rfennell/AzurePipelines#764]
Hi

I am trying to use this extension to generate release notes between the previews release and the latest release to production..

I have ticked the Allow scripts to access the OAuth token on the Agent Job however the task gives me the following error...

(node:14652) UnhandledPromiseRejectionWarning: Unable to locate access token.  Please make sure you have enabled the "Allow scripts to access OAuth token" setting.

![image](https://user-images.githubusercontent.com/68114811/87161940-8a213780-c2c5-11ea-8ec9-f9603da952c3.png)

[4_Generate Release Notes based on Release Comparison API.log](https://github.com/rfennell/AzurePipelines/files/4903446/4_Generate.Release.Notes.based.on.Release.Comparison.API.log)

**Comment by rfennell** [comment]
I have not see that before, and in my release pipeline for the task I test in both YAML and UI based pipeline.

Ticking this box exposes the OAUTH token as an environment variable for the task to pickup, so the questions why it it not present. So in the 'initialize job' step you should see something like

```
[SYSTEM_ACCESSTOKEN] --> [***]
````

![image](https://user-images.githubusercontent.com/324718/87288853-d0f47480-c4f3-11ea-9868-269f1e8f5436.png)


I am not sure what to do to help here. You could try running a release with the variable `system.debug=true` set to see if the log shows anything else.

**Comment by ChristoBothaCBS** [comment]
Good Morning.

Thank you for the feedback... The system.debug shows the following...
[SYSTEM_ENABLEACCESSTOKEN] --> [False]

![image](https://user-images.githubusercontent.com/68114811/87395150-2e9ac680-c5b1-11ea-90f4-5108f0d1a244.png)

**Comment by rfennell** [comment]
That is very strange, as I am sure you will expect my says this value is true

![image](https://user-images.githubusercontent.com/324718/87406333-7a9d3980-c5b8-11ea-9447-e9ae3ddaaf07.png)

I have never seen this before, and on searching can find no reference to the problem other than [old 2017 issues](https://developercommunity.visualstudio.com/content/problem/120616/unable-to-enable-allow-scripts-to-access-oauth-tok.html) that are meant to be fixed

All I can suggest is

1. Delete the release and try again, a long shot I admit
1. Try the task in a build to make sure that works
1. Swap to multistage YAML pipelines
1. Log an issue on https://developercommunity.visualstudio.com/spaces/21/index.html

**Comment by HaGGi13** [comment]
Unfortunately, I don't get this behavior reproduced either and have only seen similar behavior when a release was re-run and the Release Definition and not the release itself was modified to match the one already run.
The Azure DevOps/TFS - as far as I know - makes a copy of the Release Definition for execution, so no changes from the Release Definition will take effect, only those made directly in the release itself.

---

I also have a question to this subject.
Can you please explain why an OAuth token access is needed?

**Comment by rfennell** [comment]
Just to be clear I did not mean some re-run a past release, I meant to just re-try the whole process for a new release of the same release definition. But, as I said this was more hope over experiences. I did not expect any different results.

<hr>

The OAUTH checkbox setting is needed to expose the System Access Token as an environment variable which the task can pickup to use to create a credential object to access the Azure DevOps REST API with the privileges of the build/release user

```node
// get the SYSTEM_ACCESSTOKEN env. variable, not _ are replaced by . in name
var accessToken = tl.getVariable("System.AccessToken");
if (!accessToken || accessToken.length === 0) {
    throw "Unable to locate access token.  Please make sure you have enabled the \"Allow scripts to access OAuth token\" setting.";
} else {
    credHandler = webApi.getHandlerFromToken(accessToken);
}
```

This code got added a few years ago when there was a major refactor from V1 to V2 that swapped from raw REST calls to using the then VSTS Node API (now Azure DevOps Node API). Looking that current Node SDK there are other ways to get this the access token, but as this method has worked well in all other cases for a long time so I am not keen to swap it with careful consideration of the implication.

As to your problem, I have never seen this problem where ticking the OAUTH checkbox does not result in the variable being exposed. I have a good search around forums and the like and it is a strange one. 

Is it only on this release definition that you don't see the SYSTEM_ACCESSTOKEN appear when OAUTH is set for that agent job?
